### ISSUE TITLE
MathLink vs. WSTP?

### ISSUE THREAD

**Issue by stevengj** [JuliaInterop/MathLink.jl#1]
MathLink seems to be deprecated nowadays compared to the WSTP interface.   Do you have any comments on why you chose MathLink over WSTP?

cc: @simonbyrne, @rpestourie

**Comment by simonbyrne** [comment]
I think this package predates WSTP (WSTP is the updated & renamed MathLink for Mathematica 10).

Honestly, it probably makes sense to create a new package: I'll contribute where I can, but I'm leaving UCL soon, so will lose access to the institutional site licence (and I don't use it enough to justify purchasing a personal copy).

**Comment by simonbyrne** [comment]
Actually, from what I can tell, MathLink and WSTP are identical, other than the prefixes used. So it probably makes sense to keep using MathLink in order to support older versions (or make prefix flexible).

**Comment by stevengj** [comment]
That's good to hear; it would be a pain to do something new from scratch.  I wonder why they renamed it?

**Comment by simonbyrne** [comment]
I assume it was so that it has the word "Wolfram" in it.

(they also renamed the Mathematica language to "Wolfram Language" in v10)

**Comment by one-more-minute** [comment]
Yeah, WSTP wasn't around at all when I wrote this and I'd bet money that it's just a trivial rename. Presumably they'll keep supporting the `ML*` functions as aliases in which case it's a lot easier (especially for the sake of backwards compatibility) to keep things as they are.

**Comment by simonbyrne** [comment]
I think we can close this, at least until the MathLink interface is deprecated.

**Comment by simonbyrne** [comment]
Also, for what its worth, WSTP isn't included on the raspberry pi, whereas mathlink is.

**Comment by simonbyrne** [comment]
It should be possible to make it work with either WSTP or MathLink: the easiest option might be to check which symbols exist in the dynamic library.

The other question is whether we should rename the package: one candidate is WSTP.jl, however [The Julia package naming guidelines](https://julialang.github.io/Pkg.jl/v1/creating-packages/#Package-naming-guidelines-1) suggest avoiding acronyms. Another option is WolframEngine.jl?

**Comment by simonbyrne** [comment]
At the moment, I am still planning on keeping this package as minimal as possible: link handling, representation of WExprs, put/get for Julia numeric types, basic evaluation, etc.

**Comment by MasonProtter** [comment]
What if we called it something like WolframLink.jl or just Wolfram.jl? I think that'd most accurately reflect what it is without having to use an undiscoverable name like WSTP.jl?

Mathematica.jl is perhaps good enough, but I don't think this gives access to any of the features that actually differentiates Mathematica from the Wolfram Engine.

**Comment by JeffreySarnoff** [comment]
+1 WolframEngine.jl

**Comment by simonbyrne** [comment]
I think I'll register as MathLink.jl for the time being. We can see how the package evolves in terms of functionality, and perhaps rename it then.

**Comment by Roger-luo** [comment]
It looks like an FFI to mma, what about MathematicaCall.jl ?

**Comment by jebej** [comment]
I think WolframEngine makes more sense at this point, as it is free and so is what most people will be using.
